src/components/templates/Audience/Audience.scss

```diff
@@ -142,22 +151,23 @@ $seat-size: 4vh;
     flex: 1;
     flex-direction: column;
     justify-content: space-between;
-
-    padding: $spacing;
-
-    width: fit-content;
+    position: relative;
+    width: 100%;
     height: fit-content;
+    overflow: auto;
 
     .seat-row {
       display: flex;
-      justify-content: space-between;
+      justify-content: center;
     }
 
     .seat {
       display: flex;
       border-radius: 100%;
       width: $seat-size;
       height: $seat-size;
+      min-width: $seat-size;
+      min-height: $seat-size;
```

Comment:
min-width/min-height ensures that the width is going to be $seat-size, even if their parent is smaller. I guess the reason I added it is because the seats shrank when you decreased the width

https://dev.to/ziizium/the-difference-between-css-width-min-width-and-max-width-property-49ah

Reply:
Based on a quick skim through that link you provided, it doesn't sound like setting `width` and `min-width` to the same thing is going to have any difference in effect?

Reply:
I believe the main 'issue' with the sizing here is that we size the seats with `4vh`, so it's 4% of the viewport height (aka browser window height). That's why when the height is small, and the width is large, we get weird display issues.

Reply:
I believe what we actually want to do here is to use a fixed minimum width/height, probably in `px`. That would allow us to force a minimum size.

We would then also want to use that same minimum when calculating the position/size/etc of the video container within the middle 'cut out area'. If we ensure that that middle area always has enough size that the emoji area will fit, then we should be fine, regardless of screen size/weird dimensions/etc.

Reply:
removed in https://github.com/sparkletown/sparkle/pull/1237/commits/6ff3b5876858d4e306936f59a383e5f0e9c734aa

Reply:
@mike-lvov Did you read the other comments I added here? As they are directly related to ensuring the emoji container doesn't break out/etc.

Reply:
> We would then also want to use that same minimum when calculating the position/size/etc of the video container within the middle 'cut out area'. If we ensure that that middle area always has enough size that the emoji area will fit, then we should be fine, regardless of screen size/weird dimensions/etc.

I do agree that this is the best approach to handle all of the use cases, but I don't think it is time-efficient to implement this since I'm not really aware of the logic that is used for `cut out area` size. It will take a substantial amount of my time to get into this(to only handle a case with really wide and short screens)

Reply:
The logic remains the same, it calculates the rows/columns/etc. The difference is that we need to update the inline style applied to positioning our container so that it also uses the same minimum.

Reply:
Do you know exactly what to do? If so, could you apply those changes, please?

Reply:
I believe I do, was just playing with it in devtools..

I believe this is the calculation that would work for the video container part (obviously using the correct row/col vals from the code):

```
element.style {
    width: calc(20 * min(20px, 4vh));
    height: calc(15 * min(20px, 4vh));
}
```

And this is what we need to apply to `seat`, `not-seat`, and `avatar`:

```css
.audience-container .audience .seat {
    // ..snip..
    width: 4vh;
    height: 4vh;
    min-width: 20px;
    min-height: 20px;
    // ..snip..
}
```

Assuming `20px` is the value we want to use as a minimum. Basically, whatever we pick should allow enough space for the emoji reactions section even if we don't change it to make it responsive.

Reply:
Got some changes in my working copy, will commit them in a minute.

Reply:
Added in https://github.com/sparkletown/sparkle/pull/1237/commits/0beac219a5a4474071574e883b6a7b828e7af916 it's still not 100% perfect, I believe to do so we would just need to pick a minimum `px` size that would allow the number of 'carved out rows' worth of sizing to account for the non-responsive reaction container.

Reply:
@0xdevalias this has done the trick, I believe
https://github.com/sparkletown/sparkle/pull/1237/commits/958c41f091f15e3cc0b9627a6dfd86e76216578f

Reply:
See the changes I made in https://github.com/sparkletown/sparkle/pull/1237/commits/88a23c5ae42316b2bbb9d3e46b47b5fc186ba259, which I believe may make your change unneeded.

Reply:
Even at ridiculous dimensions, it all stays within the middle container now:

![image](https://user-images.githubusercontent.com/753891/110103307-e78c6a80-7df9-11eb-87af-fe279988a079.png)

![image](https://user-images.githubusercontent.com/753891/110103342-f2df9600-7df9-11eb-91ee-f715e1147ba7.png)

---

iPhone in landscape:
![image](https://user-images.githubusercontent.com/753891/110103513-21f60780-7dfa-11eb-8b74-c6bb21d005f5.png)

iPad:
![image](https://user-images.githubusercontent.com/753891/110103574-30dcba00-7dfa-11eb-960a-e6f5e2586c41.png)

Hi DPI Laptop:
![image](https://user-images.githubusercontent.com/753891/110103678-4f42b580-7dfa-11eb-8635-c6bb2b58dbeb.png)

Medium DPI Laptop:
![image](https://user-images.githubusercontent.com/753891/110103723-59fd4a80-7dfa-11eb-8940-68e2660cdefa.png)

Reply:
@0xdevalias Reverted

Reply:
Awesome, sounds good!